### ISSUE TITLE
idea: An email badge

### ISSUE THREAD

**Issue by tunnckoCore** [badgen/badgen.net#293]
Just thinking. Bots and spammers may be pretty clever nowadays, so I'm not sure if putting an email inside a url (for badge generation) is good idea too.

Fanatism isn't a bad thing some times. :D 

Badge: ![](https://badgen.net/badge/%20/consulting@tunnckocore.com?label&color=07ba96)
Source: `https://badgen.net/badge/%20/consulting@tunnckocore.com?label&color=07ba96`

So, what I'm proposing the following

```
badgen.net/email/<tld>/<name>/<domain name>
badgen.net/email/com/consulting/tunnckocore
badgen.net/email/co.uk/foobar/mydomain
```

to be able to generate badge like the above (of course with possibility for `label` and etc).

The idea of live badge will be just to combine the parts in correct order.

**Comment by tunnckoCore** [comment]
https://email-badge-m4f6gmz9pry6.runkit.sh

Badge: ![](https://badgen.net/https/email-badge-m4f6gmz9pry6.runkit.sh/co.uk/tunnckocore/matu-hiva?label)
Source: `/https/email-badge-m4f6gmz9pry6.runkit.sh/co.uk/tunnckocore/matu-hiva?label`

Badge: ![](https://badgen.net/https/email-badge-m4f6gmz9pry6.runkit.sh/com/ceo/tunnckocore?label&color=pink)
Source: `/https/email-badge-m4f6gmz9pry6.runkit.sh/email-badge-m4f6gmz9pry6.runkit.sh/com/ceo/tunnckocore?label&color=pink`

**Comment by amio** [comment]
That's a pretty nice idea!

Prefer this schema `/email/:name/:domain`

Let's use this as default style:
![](https://badgen.net/https/liam-badge-daknys6gadky.runkit.sh/bg/charlike.mike/mail)

**Comment by tunnckoCore** [comment]
I agree it's better but thinking that it's still too easy to be recognized/parsed as email. who knows what these bots do...

**Comment by amio** [comment]
Hiding email address from spammers is a road without end and it shouldn't be the goal of email badge.

While, maybe this can be a standalone api endpoint for [`/https`](https://badgen.net/docs/https), with all kinds of dark magic to hide email address.

Also, we could have a section on badgen.net, showcasing / inspiring these usage of badgen.net, that would be cool!

**Comment by tunnckoCore** [comment]
Yup. Of course it's road without end. But we can do it in that simple way.

Anyway, as you want. I'm using the `/https` for now. :)

**Comment by binyamin** [comment]
@amio I would much prefer to use a builtin endpoint, and not need to set up my own `/https` endpoint as well.

**Comment by amio** [comment]
@binyamin Do you mean a email endpoint with builtin obscuration from email parsing/recognization?

**Comment by binyamin** [comment]
@amio I meant https://github.com/badgen/badgen.net/issues/293#issuecomment-506623856

**Comment by amio** [comment]
@binyamin Nice, let's go for it 👍
